### PR TITLE
Unsilence test_rbg

### LINKED ISSUE #720: Un-silence a unit test: test_rgb.py
### Context

While working on https://github.com/ivadomed/ivadomed/pull/714, I had to silence the function `save_color_labels` because the way it was implemented produced an error when the header of the reference image was copied to the output prediction. Namely, the field "dtype" was not compatible, yielding the following error when saving the  NIFTI file:
~~~
>           raise WriterError('Cannot cast to or from non-numeric types')
E           nibabel.arraywriters.WriterError: Cannot cast to or from non-numeric types
~~~

### TODO

- Fix the implementation of `save_color_labels`
- Reestablish the calls to this function, which were removed in https://github.com/ivadomed/ivadomed/pull/714/commits/52079bd339b6e0cdffa61b778da2bae9a3560e55
- I had to silence a test (see this commit https://github.com/ivadomed/ivadomed/pull/714/commits/0f946fc3fe4b083d33783fe96a4be5e9ed207558), which was lower priority compared to the PR. This silenced test should be revisited.

### PR DESCRIPTION
**Pull request by cakester**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [ ] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

note: need to get rgb_test_file.nii.gz inside data_testing repo first

## Linked issues

resolves #720

### CODE REVIEW

**Comment by cakester** [comment]
@jcohenadad can you approve this https://github.com/ivadomed/data-testing/pull/14 for this pr? thanks

**Comment by jcohenadad** [comment]
> @jcohenadad can you approve this [ivadomed/data-testing#14](https://github.com/ivadomed/data-testing/pull/14) for this pr? thanks

done 😊

**Comment by dyt811** [comment]
The RGB_test.nii.gz file looks good. @cakester, FYI, you can use MRICRON.exe to check. 

image

Also, make sure to turn off smoothing... to look at real voxel data
image

File: testing/unit_tests/test_rgb.py [hunk 1]
```diff
@@ -2,33 +2,30 @@
...
+        image[0][2][5][i] = 1
+        image[1][2][5][i+3] = 1
+        image[2][2][5][i + 6] = 1
+    image_n = np.array(image)
+    imed_visualize.save_color_labels(
+        gt_data=image_n,
+        binarize=False,
+        gt_filename=os.path.join(
+            __data_testing_dir__,
+            "rgb_test_file.nii.gz"),
```
Comment (dyt811) [inline testing/unit_tests/test_rgb.py:25]: This input reference image I presume already has the correct header. Could you maybe briefly elaborate a bit about how that was created initially for record purposes? For the purpose of the test of "saving" functionalities, it is more than sufficient. From a test perspective, we achieved the goal if restoring the earlier test and ensure the `save_color_labels` are correctly saving the files inputs are correct.

File: ivadomed/scripts/download_data.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
             `Spine Generic <https://github.com/spine-generic/data-multi-subject>`_.
             Used for Tutorial and example in Ivadomed."""},
     "data_testing": {
-        "url": ["https://github.com/ivadomed/data-testing/archive/r20210322.zip"],
+        "url": ["https://github.com/ivadomed/data-testing/archive/r20210628.zip"],
```
Comment (dyt811) [inline ivadomed/scripts/download_data.py:25]: 👍

**Review by dyt811 — APPROVED** [review]
While I am not 100% familiar with RGB labelling in nifti and ground truth labelling, from a unit test perspective, this restored the intended goals of checking the `saving` functionality. However, the edge cases for incompatible input (such as the old ground truth file referenced), will not work and would likely require additional unit test cases to catch.  

I was a bit surprised that this function is not called anywhere else it seems. 
image except some alternative commented out path during testing phase. So we might need to revaluate its important down the road. For now, it is good enough.

File: testing/unit_tests/test_rgb.py (hunk 1 above)
Comment (cakester) [reply testing/unit_tests/test_rgb.py:25]: this new image file is created by using save_color_labels function without providing a header for the nifti image, I am just using it again to copy the header
